### PR TITLE
[metrohash] support Android triplet build

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

My team was using this library. Recently we found that build on Android reported errors for arch-specific intrinsic usage.  
This PR updates `portfile.cmake` and `CMakeLists.txt` to check variable `ANDROID` and handle the case.

`CONTROL` file is changed to `vcpkg.json` with new `"port-version"`.

### Which triplets are supported/not supported?

The following triplets should become available when the user followed docs/users/android.md correctly.

* `arm-android`
* `arm64-android`
* `x86-android`
* `x64-android`

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Wish conversion from `CONTROL` to `vcpkg.json` was correct :)

### CODE REVIEW

**Comment by luncliff** [comment]
Seems like I made a mistake with version JSON file. Will update it again after fixing `x64-osx` build.

File: port_versions/baseline.json [hunk 1]
```diff
@@ -3685,8 +3685,8 @@
       "port-version": 7
     },
     "metrohash": {
-      "baseline": "1.1.3-1",
-      "port-version": 0
+      "baseline": "v1.1.3",
```
Comment (NancyLi1013) [inline port_versions/baseline.json:3688]:
```suggestion
      "baseline": "1.1.3",
```

File: port_versions/m-/metrohash.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "715a78996ce2b3c4cf49408667ce06748297bbb1",
+      "version-string": "v1.1.3",
```
Comment (NancyLi1013) [inline port_versions/m-/metrohash.json:5]:
```suggestion
      "version-string": "1.1.3",
```

File: ports/metrohash/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "metrohash",
+  "version-string": "v1.1.3",
```
Comment (NancyLi1013) [inline ports/metrohash/vcpkg.json:3]:
```suggestion
  "version-string": "1.1.3",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: port_versions/baseline.json (hunk 1 above)
Comment (luncliff) [reply port_versions/baseline.json]: Sorry for the late update. fixed in 98743af

**Comment by NancyLi1013** [comment]
Need to run `vcpkg.exe x-add-version metrohash` to solve version check on x86-windows.

```
Found the following errors:
Error: While reading versions for port metrohash from file: C:\a\1\s\versions\m-\metrohash.json
       File declares version `1.1.3#2` with SHA: 0c764c9b22fa64b5194e65f69bfb28bc47dc20fc
       But local port with the same verion has a different SHA: fdf6869318d7d54c92233f005a39316f344a1536
       Please update the port's version fields and then run:

           vcpkg x-add-version metrohash

       to add a new version.
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the improvement!

**Pull request merged**